### ISSUE TITLE
Label merging is broken after adding current working directory in the advanced executables configuration

### ISSUE THREAD

**Issue by rafalborczuch** [matepek/vscode-catch2-test-adapter#427]
**Checklist**

- [x] The issue is about this extension and NOT about a fork.
- [x] Checked the [ALL the SUPPORT](https://github.com/matepek/vscode-catch2-test-adapter/blob/master/documents/support.md) document.
- [x] The latest version of the extension was used.
- [x] It is not related to remote-vscode or I checked the following [issue](https://github.com/matepek/vscode-catch2-test-adapter/issues/201)
- [x] Imagine yourself into my position and think how hard to debug the issue without insufficient information.
      I understand that you have privacy concerns and I expect you to understand that this extension is developed for free.
      Thanks.

**Describe the bug**

Label merging stops working after adding `cwd` property to the advanced executables configuration.
The `cwd` path exists on the filesystem and tests have the correct cwd set when running/debugging.
I use a multi-root workspace.

**To Reproduce**

The following config correctly merges the labels in the tree view. Note that the `cwd` property is commented.
```js
"testMate.cpp.test.advancedExecutables": [
    {
        "pattern": "${command:cmake.activeFolderName}/bin/linux/${command:cmake.buildType}/**/*{test,Test,TEST}*",
        // "cwd": "${relDirpath}/ctest_cwds/${filename}",
        "gtest": {
            "testGrouping": {
                "groupByExecutable": {
                    "label": "${absDirpath[-4:-3]}",
                    "description": "${relDirpath}",
                    "mergeByLabel": true,
                    "groupByExecutable": {
                        "label": "${filename}",
                        "description": "${workspaceName}",
                        "groupByTags": {
                            "tags": [],
                            "tagFormat": "${tag}"
                        }
                    },
                }
            }
        }
    }
],
```

producing the following tree view in the Test Explorer:
```
projectA
    executable1
        TestClassA
            testA1
            tetsA2
        TestClassB
            testB1
            tetsB2
    executable2
        TestClassC
            testC1
            tetsC2
        TestClassD
            testD1
            tetsD2
```

After uncommenting the `cwd` line:
```js
"testMate.cpp.test.advancedExecutables": [
    {
        "pattern": "${command:cmake.activeFolderName}/bin/linux/${command:cmake.buildType}/**/*{test,Test,TEST}*",
        "cwd": "${relDirpath}/ctest_cwds/${filename}",
        "gtest": {
            "testGrouping": {
                "groupByExecutable": {
                    "label": "${absDirpath[-4:-3]}",
                    "description": "${relDirpath}",
                    "mergeByLabel": true,
                    "groupByExecutable": {
                        "label": "${filename}",
                        "description": "${workspaceName}",
                        "groupByTags": {
                            "tags": [],
                            "tagFormat": "${tag}"
                        }
                    },
                }
            }
        }
    }
],
```

tree view in the Test Explorer doesn't merge 1st level label:
```
projectA
    executable1
        TestClassA
            testA1
            tetsA2
        TestClassB
            testB1
            tetsB2
projectA
    executable2
        TestClassC
            testC1
            tetsC2
        TestClassD
            testD1
            tetsD2
```

Each executable has its own 1st level node (`projectA` in the example).

**Desktop**

- Extension Version: v4.9.0
- VS Code Version: 1.85.2
- Catch2 / Google Test / DOCTest Version: Google Test 1.6
- OS Type and Version: Windows 11 23H2
- Using remote-ssh/docker/wsl?: Using remote dev container (CentOS 7) with WSL 2 (Ubuntu 22.04)

**Comment by matepek** [comment]
where is the `ProjectA` coming from?

**Comment by rafalborczuch** [comment]
The above assumes that the directory structure is as follows:
```
repo
    projectA
    projectB
    repo.code-workspace
```
and it is opened in the VS Code as a mutli-root workspace file `repo.code-workspace` that has the following folders defined:
```
"folders": [
    {
        "path": "projectA"
    },
    {
        "path": "projectB"
    }
]
```
The build output binaries are placed under `projectA/bin/linux/Debug` and `projectB/bin/linux/Debug` and the `"label": "${absDirpath[-4:-3]}",` results in `projectA` or `projectB`.

**Comment by matepek** [comment]
And what is the expectations for that label?
It's a but unclear what is the current value and what you expect.
Have you tried using `${relDirpath[0]}`?

**Comment by rafalborczuch** [comment]
`${relDirpath[0]}` gives the same result.
The expected top level label is `projectA` or `projectB`.

**Comment by rafalborczuch** [comment]
The problem is here: https://github.com/matepek/vscode-catch2-test-adapter/blob/master/src/framework/AbstractExecutable.ts#L293
The `this.shared.options` has `cwd` which has different value for each executable (due to the `advancedExecutable` config `"cwd": "${relDirpath}/ctest_cwds/${filename}"`) resulting in different options hash value that is used to construct the `id`.
The `mergeByLabel` configuration works more like `mergeByEnvAndCwd`.

**Comment by matepek** [comment]
is `${relDirpath}/ctest_cwds/${filename}` a directory or a file?

**Comment by matepek** [comment]
<details>
<summary>Fixed in <b>v4.11.0</b>.</summary>

This issue was mentioned in [CHANGELOG.md](./CHANGELOG.md) under a released entry so it is assumed to be fixed.
User verifications are always welcome.
</details>

**Comment by rafalborczuch** [comment]
Thanks, I have successfully verified the fix.
